packages/e2e-tests/plugins/interactive-blocks/directive-on-document/render.php

```diff
@@ -0,0 +1,15 @@
+<?php
+/**
+ * HTML for testing the directive `data-wp-on`.
+ *
+ * @package gutenberg-test-interactive-blocks
+ */
+
+gutenberg_enqueue_module( 'directive-on-document-view' );
+?>
+
+<div data-wp-interactive='{ "namespace": "directive-on-document" }'>
+	<div data-wp-on-document--keydown="callbacks.keydownHandler">
```

Comment:
Can you test that when this element is removed from the DOM, the event listener is also removed?

As we don't have ways to remove elements from the DOM yet, what we are doing is wrapping them in a fake `wp-show` directive: https://github.com/WordPress/gutenberg/blob/d855787481f9f27ffec3a1113bf68bec22f7dbff/packages/e2e-tests/plugins/interactive-blocks/directive-init/view.js#L6-L17

Reply:
Done in https://github.com/WordPress/gutenberg/pull/57931/commits/99e2ed82f985caff54a8a09745b91fca39241464

Reply:
Cool.

Actually, we could also check that the event listener is reattached correctly:

```js
await page.keyboard.press("ArrowDown");
await expect(counter).toHaveText("1");
// Remove the element.
await visibilityButton.click();
// This keyboard press should not increase the counter.
await page.keyboard.press("ArrowDown");
// Add the element back.
await visibilityButton.click();
await expect(counter).toHaveText("1");

// ↓ This ↓
// Check that the event listener is reattached.
await page.keyboard.press("ArrowDown");
await expect(counter).toHaveText("2");
```

Reply:
I did, and remove it 😆 , as I thought I was repeating too much tests.
I will add it again.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/57931/commits/b9c83d1be6fae670ac81ba7a4f66e74e4f6c5adc

Reply:
I think it's good to cover re-attachment, thanks 🙂